frontend/stmt_inferrer.py

```diff
@@ -0,0 +1,296 @@
+"""Inferrer for python statements.
+
+Infers the types for the following expressions:
+    - Assign(expr* targets, expr value)
+    - AugAssign(expr target, operator op, expr value)
+    - Return(expr? value)
+    - Delete(expr* targets)
+    - If(expr test, stmt* body, stmt* orelse)
+    - While(expr test, stmt* body, stmt* orelse)
+    - For(expr target, expr iter, stmt* body, stmt* orelse)
+    - AsyncFor(expr target, expr iter, stmt* body, stmt* orelse)
+    - With(withitem* items, stmt* body)
+    - AsyncWith(withitem* items, stmt* body)
+    - Try(stmt* body, excepthandler* handlers, stmt* orelse, stmt* finalbody)
+
+    TODO:
+    - Import(alias* names)
+    - ImportFrom(identifier? module, alias* names, int? level)
+    - Global(identifier* names)
+    - Nonlocal(identifier* names)
+"""
+
+import expr_inferrer as expr, ast, sys
+from context import Context
+from i_types import *
+
+def __infer_assignment_target(target, context, value_type):
+    """Infer the type of a target in an assignment
+
+    Attributes:
+        target: the target whose type is to be inferred
+        context: the current context level
+        value_type: the type of the value assigned to the target
+
+    Target cases:
+        - Variable name. Ex: x = 1
+        - Tuple. Ex: a, b = 1, "string"
```

Comment:
I just remembered, you can have starred expressions as targets. E.g. 
a, *b, c = e